src/NHapi.Base/SupportClass.cs

```diff
@@ -1,3268 +1,3486 @@
+        void startElement(string namespaceURI, string localName, string qName, SaxAttributesSupport atts);
+
+        /// <summary>
+        /// This methods indicates the start of a prefix area in the XML document.
+        /// </summary>
+        /// <param name="prefix">The prefix of the area.</param>
+        /// <param name="uri">The namespace URI of the prefix area.</param>
+        void startPrefixMapping(string prefix, string uri);
+    }
+
+    /*******************************/
+
+    /// <summary>
+    /// This interface is created to emulate the SAX Locator interface behavior.
+    /// </summary>
+    public interface IXmlSaxLocator
+    {
+        /// <summary>
+        /// This method return the column number where the current document event ends.
+        /// </summary>
+        /// <returns>The column number where the current document event ends.</returns>
+        int getColumnNumber();
+
+        /// <summary>
+        /// This method return the line number where the current document event ends.
+        /// </summary>
+        /// <returns>The line number where the current document event ends.</returns>
+        int getLineNumber();
+
+        /// <summary>
+        /// This method is not supported, it is included for compatibility.
+        /// </summary>
+        /// <returns>The saved public identifier.</returns>
+        string getPublicId();
+
+        /// <summary>
+        /// This method is not supported, it is included for compatibility.
+        /// </summary>
+        /// <returns>The saved system identifier.</returns>
+        string getSystemId();
+    }
+
+    /*******************************/
+
+    /// <summary>
+    /// This class is created for emulates the SAX LocatorImpl behaviors.
+    /// </summary>
+    public class XmlSaxLocatorImpl : IXmlSaxLocator
+    {
+        /// <summary>
+        /// This method returns a new instance of 'XmlSaxLocatorImpl'.
+        /// </summary>
+        /// <returns>A new 'XmlSaxLocatorImpl' instance.</returns>
+        public XmlSaxLocatorImpl()
+        {
+        }
+
+        /// <summary>
+        /// This method returns a new instance of 'XmlSaxLocatorImpl'.
+        /// Create a persistent copy of the current state of a locator.
+        /// </summary>
+        /// <param name="locator">The current state of a locator.</param>
+        /// <returns>A new 'XmlSaxLocatorImpl' instance.</returns>
+        public XmlSaxLocatorImpl(IXmlSaxLocator locator)
+        {
+            setPublicId(locator.getPublicId());
+            setSystemId(locator.getSystemId());
+            setLineNumber(locator.getLineNumber());
+            setColumnNumber(locator.getColumnNumber());
+        }
+
+        /// <summary>
+        /// This method is not supported, it is included for compatibility.
+        /// Return the saved public identifier.
+        /// </summary>
+        /// <returns>The saved public identifier.</returns>
+        public virtual string getPublicId()

```

Comment:
make thies methods properties? and mark methods as obsolete?

Reply:
will log a separate issue to refactor this `SupportClass`